### PR TITLE
[1.9.4] Introduced Capabilities for World

### PR DESCRIPTION
**Pull request by Abastro**
This will allow compatibility between mods in World part.
(PR related with my suggestion #2877, note that server part is not included here)
+ Closing previous PR since changing target branch is not allowed.

### CODE REVIEW

File: src/main/java/net/minecraftforge/world/WorldCapabilityData.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        super(fileNameForProvider(provider));
+        this.serializable = serializable;
+        this.dimension = provider.getDimension();
+    }
+
+	@Override
+    public void readFromNBT(NBTTagCompound nbt)
+    {
+	    if(nbt.hasKey("ForgeCaps"))
```
Comment (kashike) [inline src/main/java/net/minecraftforge/world/WorldCapabilityData.java]: Don't worlds already have a `Forge` compound tag? If so, why not add a `Capabilities` compound under the `Forge` compound?

Comment (Abastro) [inline src/main/java/net/minecraftforge/world/WorldCapabilityData.java]: @kashike As far as I know, There is no Forge compound tag for world itself.

Comment (williewillus) [inline src/main/java/net/minecraftforge/world/WorldCapabilityData.java]: Also better to be ForgeCaps because that's what all the other caps do

**Comment by Actuarius** [comment]
@tterrag1098 added labels [Feature]

**Comment by Abastro** [comment]
Is there any issues with this PR?
On the designs, implementations, or anything.
If any, please let me know.

**Comment by cpw** [comment]
I like this idea. We do have world mods that do some hacking of world state.

**Comment by cpw** [comment]
Closing in favour of #3069

**Pull request closed without merging**